SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -60,12 +74,12 @@ public override bool Execute()
             pi.ProjectName = this.ProjectName;
             pi.FullPath = this.FullProjectPath;
 
-            // TODO: handle failures and missing values.
             Guid projectId;
             if (Guid.TryParse(this.ProjectGuid, out projectId))
             {
                 pi.ProjectGuid = projectId;
                 pi.AnalysisResults = TryCreateAnalysisResults(this.AnalysisResults);
+                pi.AnalysisSettings = TryCreateAnalysisSettings(this.AnalysisSettings);
```

Comment:
Unsolicited opinion here. I tend to agree with duncan. I only use the bool return with the out when null could be a valid return value. Using the out I end up suppressing the FxCop rule that tells about not using out parameters.